ar_preproc_node:
  ros__parameters:
    tracker_frame: 'map'
    ar_tag_ids: [20]
    labels: ['spacecraft']

tbd_tracker_node:
  ros__parameters:
    tracker:
      frame_id: 'map'
      mismatch_penalty: 100.
      assignment_algo: 'greedy'
      yaw_corr: False
      publishers:
          names: ['track_pub', scene_pub]
          track_pub: # Publisher name
            pub_topic: 'tracks'
            msg_type: 'Tracks3D'
            function: 'publish_tracks'
            queue_size: 10
          scene_pub: # Publisher name
            pub_topic: 'scenes'
            msg_type: 'SceneUpdate'
            function: 'publish_scene'
            queue_size: 10

    object_properties: 
      object_classes: ["spacecraft"]
      spacecraft:
          model_type: 'cvcy_obj'
          yaw_proc_var: [1.]
          size_proc_var: [1., 1., 1.]
          vel_proc_var: [1., .5, .5]
          length: 0.25
          width: 0.25
          height: 0.5
          create_method: 'conf'
          delete_method: 'conf'
          active_thresh: .25
          delete_thresh: .15
          detect_thresh: .15
          score_update_function: 'parallel_add'
          score_decay: .001
          sim_metric: 'dist_3d'
          match_thresh: 0.5
    
    detectors:
      #Declare names of detectors upfront
      detector_names: ['artag']

      # For each named detector, give detector model information
      artag:
        topic: 'converted_detections'
        msg_type: 'Detections3D'
        detector_type: 'pos_bbox_3d'
        detection_classes: ["spacecraft"]
        detection_properties:
            spacecraft:
                pos_obs_var: [1., 1., 1.]
                yaw_obs_var: [1.]
                size_obs_var: [1., 1., 1.]
                object_class: 'spacecraft'
                ignore: False
            